src/CRSMatrix.hpp

```diff
@@ -119,10 +122,48 @@ class CRSMatrix : protected CRSMatrixView< T, COL_TYPE, INDEX_TYPE >
 
   /**
    * @brief Default move assignment operator, performs a shallow copy.
+   * @param src The CRSMatrix to be moved from.
    * @return *this.
    */
   inline
-  CRSMatrix & operator=( CRSMatrix && ) = default;
+  CRSMatrix & operator=( CRSMatrix && src )
+  {
+    ParentClass::free( m_entries );
+    ParentClass::operator=( std::move( src ) );
+    return *this;
+  }
+
+  /**
+   * @brief Steal the resources from a SparsityPattern.
+   * @param src the SparsityPattern to convert.
+   */
+  inline
+  void stealFrom( SparsityPattern< COL_TYPE, INDEX_TYPE > && src )
```

Comment:
Lets you create fill a `SparsityPattern` and the construct a `CRSMatrix` from it.

Reply:
This is nice....but it would be cool to call this function `assimilate`.